framework/decode/openxr_replay_consumer_base.cpp

```diff
@@ -554,20 +569,26 @@ void OpenXrReplayConsumerBase::Process_xrCreateApiLayerInstance(
     };
 
     // Remove the original XR_TYPE_INSTANCE_CREATE_INFO_ANDROID_KHR structure with incorrect info
-    // TODO: This breaks if there is any structure in front of this in the next call chain.
-    const XrBaseInStructure* next_struct = reinterpret_cast<const XrBaseInStructure*>(create_info->next);
-    while (next_struct != nullptr && next_struct->type == XR_TYPE_INSTANCE_CREATE_INFO_ANDROID_KHR)
+    XrBaseInStructure*       last_next_struct = reinterpret_cast<XrBaseInStructure*>(&init_android);
+    const XrBaseInStructure* next_struct      = reinterpret_cast<const XrBaseInStructure*>(create_info->next);
+    while (next_struct != nullptr)
     {
-        next_struct = next_struct->next;
+        // Skip any old Android create info structs
+        if (next_struct->type == XR_TYPE_INSTANCE_CREATE_INFO_ANDROID_KHR)
+        {
+            last_next_struct->next = next_struct->next;
+        }
+        last_next_struct = const_cast<XrBaseInStructure*>(next_struct);
+        next_struct      = next_struct->next;
     }
-    init_android.next = reinterpret_cast<const void*>(next_struct);
-
-    XrInstanceCreateInfo new_create_info = *create_info;
-    new_create_info.next                 = &init_android;
-    create_info                          = &new_create_info;
+    init_android.next         = reinterpret_cast<const void*>(next_struct);
```

Comment:
this is just setting init_android.next to nullptr, but we already updated it by using it as the first last_next_struct

Reply:
Not if there is no Android create info.  But I'll see what I can do to clean it up.

Reply:
It is now cleaned up.